In [ ]:
import pandas as pd

intro text into what we're about to do

In [23]:
# read in data from local file path, print out first 5 rows

data = pd.read_csv("data_sets/lab_1_data.csv", header=None)
print data.head(n=5)

         0  1      2      3       4       5        6        7       8   \
0    842302  M  17.99  10.38  122.80  1001.0  0.11840  0.27760  0.3001   
1    842517  M  20.57  17.77  132.90  1326.0  0.08474  0.07864  0.0869   
2  84300903  M  19.69  21.25  130.00  1203.0  0.10960  0.15990  0.1974   
3  84348301  M  11.42  20.38   77.58   386.1  0.14250  0.28390  0.2414   
4  84358402  M  20.29  14.34  135.10  1297.0  0.10030  0.13280  0.1980   

        9    ...        22     23      24      25      26      27      28  \
0  0.14710   ...     25.38  17.33  184.60  2019.0  0.1622  0.6656  0.7119   
1  0.07017   ...     24.99  23.41  158.80  1956.0  0.1238  0.1866  0.2416   
2  0.12790   ...     23.57  25.53  152.50  1709.0  0.1444  0.4245  0.4504   
3  0.10520   ...     14.91  26.50   98.87   567.7  0.2098  0.8663  0.6869   
4  0.10430   ...     22.54  16.67  152.20  1575.0  0.1374  0.2050  0.4000   

       29      30       31  
0  0.2654  0.4601  0.11890  
1  0.1860  0.2750  0.08902  
2  0.

according to data scource information the first two columns represent patient ID and 
Diagnosis and the following columns (3-32) represent three types of  measurments for the following features:
    'radius',
    'texture',
    'perimeter',
    'area',
    'smoothness',
    'compactness',
    'concavity',
    'concave_points',
    'symmetry',
    'fractal_dimension'
   
Create the column names with the above information


In [24]:
# initialize column list, add identity, diagonosis and feature measurement columns

data_set_columns = []

identity_and_diagnosis = ['id', 'diagnosis']

feature_types = [
    'radius',
    'texture',
    'perimeter',
    'area',
    'smoothness',
    'compactness',
    'concavity',
    'concave_points',
    'symmetry',
    'fractal_dimension'
]

measurement_types = [
    'mean',
    'standard_error',
    'worst'
]

#add the identity columns
data_set_columns.extend(identity_and_diagnosis)

#add the feature columns for each measurement type
for measurement in measurement_types:
    for feature in feature_types:
        data_set_columns.append('_'.join([feature, measurement]))    
        
# set column names for breast cancer data set
data.columns = data_set_columns
print data.describe()

                 id  radius_mean  texture_mean  perimeter_mean    area_mean  \
count  5.690000e+02   569.000000    569.000000      569.000000   569.000000   
mean   3.037183e+07    14.127292     19.289649       91.969033   654.889104   
std    1.250206e+08     3.524049      4.301036       24.298981   351.914129   
min    8.670000e+03     6.981000      9.710000       43.790000   143.500000   
25%    8.692180e+05    11.700000     16.170000       75.170000   420.300000   
50%    9.060240e+05    13.370000     18.840000       86.240000   551.100000   
75%    8.813129e+06    15.780000     21.800000      104.100000   782.700000   
max    9.113205e+08    28.110000     39.280000      188.500000  2501.000000   

       smoothness_mean  compactness_mean  concavity_mean  concave_points_mean  \
count       569.000000        569.000000      569.000000           569.000000   
mean          0.096360          0.104341        0.088799             0.048919   
std           0.014064          0.052813     

essentially we have 10 features per patient and three different measurments for each feature. We can see there are 569 total 
some more information about what this data represents and a bar plot of the two categories

In [25]:
data.to_csv("data_sets/data_with_column_names.csv", index=False)